astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -445,6 +445,10 @@ def import_module(
             # skip here
             if relative_only:
                 raise
+            # Don't repeat the same operation, e.g. for missing modules
+            # like "_winapi" or "nt" on POSIX systems.
+            if modname == absmodname:
```

Comment:
Is there a difference between `absmodname` and `use_cache==False` and using `modname`? That would be a change in behaviour now.

Reply:
This occurred to me also. Looks like we missed this in #1747 (we should have added use_cache=use_cache in the final line of the function). Are you okay with me making that in change in this PR or would you like a separate one?

Reply:
Let's do it in this one. This is fairly straightforward so should be fine to increase scope a little!